climada/hazard/storm_europe.py

```diff
@@ -199,6 +206,209 @@ def _read_one_nc(self, file_name, centroids):
+                of the cosmo model
+            event_date (datetime, optional): one day within the forecast
+                period, only this day (00H-24H) will be included in the hazard
+            model_name (str,optional): select the name of the icon model to
+                be downloaded. Must match the url string
+                (see _download_icon_grib for further info)
+            description (str, optional): description of the events, defaults
+                to 'cosmo weather forecast'
+        """
+        self.clear()
+        # create centroids
+        self.centroids = self._centroids_from_nc(fp_file)
+
+        # read intensity from file
+        ncdf = xr.open_dataset(fp_file)
+        ncdf = ncdf.assign_coords(date=('time',ncdf["time"].dt.floor("D")))
+
+        if event_date:
+            # stacked = np.max(ncdf.sel(time=event_date.strftime('%Y-%m-%d')).VMAX_10M,
+            #                  axis=0).stack(intensity=('y_1', 'x_1'))
+            stacked2 = ncdf.sel(time=event_date.strftime('%Y-%m-%d')
+                                ).groupby('date'
+                                          ).max().stack(intensity=('y_1',
+                                                                   'x_1'))
+            considered_dates = np.datetime64(event_date)
+        else:
+            time_covered_step = ncdf['time'].diff('time')
+            time_covered_day = time_covered_step.groupby('date').sum()
+            # forecast run should cover at least 18 hours of a day
+            days_to_consider = time_covered_day > np.timedelta64(18,'h') 
+            stacked2 = ncdf.groupby('date'
+                                    ).max().sel(date=days_to_consider
+                                                ).stack(intensity=('y_1',
+                                                                   'x_1'))
+            unique_dates = ncdf['time'].groupby('date').min()
+            considered_dates = unique_dates.sel(date=days_to_consider).values
+
+
+        stacked2 = stacked2.stack(event=('date', 'epsd_1'))
+        stacked2 = stacked2.where(stacked2.VMAX_10M > self.intensity_thres)
+        stacked2 = stacked2.fillna(0)
+
+        # fill in values from netCDF
+        self.intensity = sparse.csr_matrix(stacked2.VMAX_10M.T)
+        self.event_id = np.arange(stacked2.date.size)+1
+
+
+        # fill in default values
+        self.units = 'm/s'
+        self.fraction = self.intensity.copy().tocsr()
+        self.fraction.data.fill(1)
+        self.orig = np.ones_like(self.event_id)*False
+        self.orig[(stacked2.epsd_1 == 0).values] = True
+        self.date = np.repeat(
+            np.array(datetime64_to_ordinal(considered_dates)),
+            ncdf.epsd_1.size
+            )
+        self.event_name = [date_i + '_ens' + str(ens_i)
+                           for date_i, ens_i in zip(date_to_str(self.date),
+                                                    stacked2.epsd_1.values+1)
+                           ]
```

Comment:
does this make `event_name` an array of length `min([len(self.date), len(stacked2.epsd_1)])`?
i'd imagine we want all combinations of date and ensemble number?